### PR TITLE
feat: add ability to define inline prompts in yaml

### PR DESCRIPTION
**Pull request by acoola**

### CODE REVIEW

File: examples/dag/dag.yaml [hunk 1]
```diff
@@ -34,7 +30,11 @@ nodes:
     name: OpenAI-1
     model: gpt-3.5-turbo
     connection: openai-conn
-    prompt: openai-ai-prompt
+    prompt:
+      id: openai-ai-prompt
```
Comment (delamainer) [inline examples/dag/dag.yaml:34]: Is `id` required?

**Review by delamainer — APPROVED** [review]

File: examples/dag/dag.yaml (hunk 1 above)
Comment (delamainer) [reply examples/dag/dag.yaml:34]: I don't think we need an `id` in inline prompt.

File: examples/dag/dag.yaml [hunk 2]
```diff
@@ -7,10 +7,6 @@ connections:
     api_key: ${oc.env:ANTHROPIC_API_KEY}
 
 prompts:
-  openai-ai-prompt:
```
Comment (vitalii-dynamiq) [inline examples/dag/dag.yaml:10]: keep both examples as we support both

**Review by olbychos — APPROVED** [review]

File: examples/dag/dag.yaml (hunk 1 above)
Comment (acoola) [reply examples/dag/dag.yaml:34]: It's not required, we autogenerate in case of a miss

File: examples/dag/dag.yaml (hunk 2 above)
Comment (acoola) [reply examples/dag/dag.yaml:10]: I left other 3 prompts in `prompts` section, but this one removed as it used as inline prompt

**Pull request merged**